config/filter.d/screensharingd.conf

```diff
@@ -0,0 +1,33 @@
+# Fail2Ban configuration file
+#
+# Author: Simon Brown
+#
+# $Revision: 1 $ 
+#
+# Filter for Mac OS X Screen Sharing service
+
+[INCLUDES]
+
+# Read common prefixes. If any customizations available -- read them from
+# common.local
+before = common.conf
+
+
+[Definition]
+
+_daemon = screensharingd
+
+# Option:  failregex
+# Notes.:  regex to match the password failures messages in the logfile. The
+#          host must be matched by a group named "host". The tag "<HOST>" can
+#          be used for standard IP/hostname matching and is only an alias for
+#          (?:::f{4,6}:)?(?P<host>[\w\-.^_]+)
+# Values:  TEXT
+#
+failregex = ^.+ screensharingd.+: Authentication: FAILED :: User Name: .+ :: Viewer Address: <HOST> :: Type: DH$
```

Comment:
__prefix_line contains platform-independent standard prefix-expressions for syslog-lines, like hostname, daemon name, pid etc.
That will be automatically set from fail2ban in accordance with your distribution. 

And as I already said `^.+` makes no sense and can be removed (to parse anchored from end of line only) or even replaced with prefix_line.